core/ecschema-locaters/src/SchemaFileLocater.ts

```diff
@@ -162,7 +162,11 @@ export abstract class SchemaFileLocater {
    * @param format The type of file that the schema key refers to. json or xml
    */
   private addCandidateSchemaKeys(foundFiles: FileSchemaKey[], schemaPath: string, fileFilter: string, desiredKey: Readonly<SchemaKey>, matchType: SchemaMatchType, format: string) {
-    const fullPath = path.join(schemaPath, fileFilter);
+    let fullPath = path.join(schemaPath, fileFilter);
+    // `glob.GlobSync` doesn't work correctly if the Glob pattern has the long path prefix for Windows systems
+    if (fullPath.startsWith("\\\\?\\")) {
```

Comment:
I took a look at the Glob docs and they make pretty clear that they do not support \ in paths (unless you set an option to force replace \\ with /)

They also seem to say they support UNC paths so long as you use //? not \\?

https://www.npmjs.com/package/glob#Windows

Further more we are using glob 7.1.2 which was released 7 years ago, the newest 7.x version was released 3 years ago (7.2.0).  Currently the latest is 10.x in the docs there is a note that v7 has buggy windows path support, search for `windowsPathsNoEscape`.  So I think the correct thing to do is to investigate updating iTwin.js to 10.x and investigating the various flags (like windowsPathsNoEscape) to support windows paths.

Stripping out the \\? really feels like a hack

Reply:
In my previous attempt, I replaced all backslashes `\\` with forward slashes `/` because glob patterns use forward slashes for path separation. This didn't work. I also tried using the setting `windowsPathsNoEscape: true` with glob v7.1.2, but it wasn't effective either. To move forward, I'll update glob and test various options.

Reply:
The glob package has been upgraded to its newest version (10.3.12). Since the updated API no longer supports `glob.sync`, references are changed to `globSync`. Additionally, all `\\` in glob patterns has been replaced with `/` as glob patterns should always use `/` as a path separator, even on Windows systems.

References:
https://github.com/isaacs/node-glob/issues/493
https://stackoverflow.com/questions/55512907/node-glob-sync-returns-empty-array